### PR TITLE
Introduction problem set

### PR DESCRIPTION
**Pull request by amirhassank**
## Changes

Added 3 new introductory problem sets

### CODE REVIEW

**Comment by pandasa123** [comment]
Hey @amirhassank, could you combine these into a single `quantum_states_and_operations.ipynb` file? @vabarbosa has created a smart way to have related problems living in one file while still having completions tracked. If we want different pages, use `##` instead

File: notebooks/problem-sets/Quantum states and operations/01.Single_qubit_quantum_states.ipynb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d7613bc7",
+   "metadata": {},
+   "source": [
+    "## Problem 1\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"test-problem\") -->\n",
```
Comment (vabarbosa) [inline notebooks/problem-sets/Quantum states and operations/01.Single_qubit_quantum_states.ipynb:10]: @amirhassank the `goal` should be a unique value. it appears you are using `test-problem` multiple times

File: notebooks/problem-sets/Quantum states and operations/02.Single_qubit_quantum_gate.ipynb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d7613bc7",
+   "metadata": {},
+   "source": [
+    "## Problem 2\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"test-problem\") -->\n",
```
Comment (vabarbosa) [inline notebooks/problem-sets/Quantum states and operations/02.Single_qubit_quantum_gate.ipynb:10]: the `goal` should be a unique value. it appears you are using `test-problem` multiple times

**Review by vabarbosa — CHANGES_REQUESTED** [review]
@amirhassank along with above comment, you should also update the `goal` values. they should be unique

**Comment by pandasa123** [comment]
@amirhassank, I believe the original folder needs to be deleted

File: notebooks/problem-sets/quantum_states_and_operations.ipynb [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+   "cell_type": "markdown",
+   "id": "b09a3194",
+   "metadata": {},
+   "source": [
+    "## Problem 4\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"quantum-states-and-operators-4\") -->\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Which of the following operator is Hermitian?\n",
```
Comment (frankharkins) [inline notebooks/problem-sets/quantum_states_and_operations.ipynb:128]:
```suggestion
    "Which of the following operators is Hermitian?\n",
```

**Review by frankharkins — CHANGES_REQUESTED** [review]
Thanks, I've made a couple of suggestions but otherwise everything looks good. You've still got some conflicting goal names (`quantum-states-and-operators-4`) which you'll need to change to pass the checks.

**Comment by pandasa123** [comment]
@amirhassank, is this ready to be reviewed?

**Comment by eddybrando** [comment]
Hi @amirhassank, we'd need you to please sign the CLA: https://github.com/Qiskit/platypus/pull/1537#issuecomment-1240725637 🙂

**Pull request closed without merging**